Extra features

ZeroTzu opened the pull request:
This PR adds
- Scanning documents directly from the user's camera
- Sorting summaries in SummaryFragment by (i.e This week, Last 3 weeks, Last 3 Months, All Time) and by Ascending/Descending Order
- Changed colours to be more visible

ZeroTzu committed:
Added Camera Feature

ZeroTzu committed:
Added Sorting Summaries feature and fixed coloring for SummaryFragment

ZeroTzu committed:
Edited Behaviour of sort button. Changed colours to be more visually appealing in all colour modes

arashnrim commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,51 +1,62 @@
+    <string name="recentActivityLabel">Recent Activity</string>
+    <string name="settings">Settings</string>
     <string name="settings_description">Change various settings for the app here.</string>
+    <string name="simplify_camera">Take a picture.</string>
```
Since this is used in a button, maybe we can get rid of the full stop at the end so it fits in with everything else?

```suggestion
    <string name="simplify_camera">Take a picture</string>
```

ZeroTzu replied:
Temporary fix to the issue, declared 
`    <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" />`
    in AndroidManifest

arashnrim reviewed (changes requested):
I'm getting a prob on my phone where I can't take a picture, all I'm seeing in the Logcat is this:

> No app to handle camera intent

Is this an Android version problem or something? My phone's using Android 11, so it should still be recent enough to not have a problem I think :eyes:

ZeroTzu committed:
Temporary fix to camera feature not working.
The the issue was that the camera app on some phone are not responding to the specified queries. So the current fix is to query all the packages that are installed by the user

ZeroTzu committed:
Changed strings.xml

arashnrim reviewed (approved):
Looks good! I think just one minor thing: it's hard to move the overlay around because it's just that one area in the top left. Can we make it have a more prominent placing or something?

ZeroTzu committed:
Made drag button more prominent and easier to tap

ZeroTzu commented:
Alright, it should look much better now
